From 1e74c8d7b194ac13dd3d5016b6b5117829beb3b9 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Fri, 06 May 2022 08:02:43 +0300
Subject: [PATCH] [SW-0] gaudi3: skip spi gw config for onix

Onix image doesn't implement D2D SPI GW registers which causes ARC
exception when accessed. This patch skips D2D SPI GW disable for
onix.

Change-Id: Iac4180981823260f294c42986a6bd87764eb8e7d
---

diff --git a/arch/gaudi3/arc/asic.c b/arch/gaudi3/arc/asic.c
index f51b029..af0ac9e 100644
--- a/arch/gaudi3/arc/asic.c
+++ b/arch/gaudi3/arc/asic.c
@@ -3,3 +3,10 @@
  * Copyright (C) 2022 HabanaLabs Ltd.
  * All Rights Reserved.
  */
+
+#include <dev_common.h>
+
+void arcboot_rom_disable_spi_gw(void)
+{
+	disable_spi_gw();
+}
diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index d1b6cbf..5a6d64f 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -436,7 +436,7 @@
 	}
 }
 
-void arcboot_rom_disable_spi_gw(void)
+void disable_spi_gw(void)
 {
 	uint32_t val;
 
@@ -448,7 +448,6 @@
 	val &= ~(D0_D2D_SPI_GLOBAL_CFG_MSTR_ENABLE_M);
 	regmap_writel(val, mmD0_D2D_SPI_GLOBAL_CFG);
 }
-
 
 void arcboot_rom_request_cpld_wdog(void)
 {
diff --git a/arch/gaudi3/arc/dev_common.h b/arch/gaudi3/arc/dev_common.h
index da7e00c..17009e3 100644
--- a/arch/gaudi3/arc/dev_common.h
+++ b/arch/gaudi3/arc/dev_common.h
@@ -13,3 +13,4 @@
 void arcboot_rom_wait_for_fuse_data(void);
 int arcboot_rom_fuse_single_die_config(void);
 void ext_rom_load_erom_image(enum hl_fw_img_type_erom img_type);
+void disable_spi_gw(void);
diff --git a/arch/gaudi3/arc/pldm.c b/arch/gaudi3/arc/pldm.c
index f51b029..af0ac9e 100644
--- a/arch/gaudi3/arc/pldm.c
+++ b/arch/gaudi3/arc/pldm.c
@@ -3,3 +3,10 @@
  * Copyright (C) 2022 HabanaLabs Ltd.
  * All Rights Reserved.
  */
+
+#include <dev_common.h>
+
+void arcboot_rom_disable_spi_gw(void)
+{
+	disable_spi_gw();
+}
